src/tests/ftest/server/metadata.py

```diff
@@ -307,77 +262,279 @@ def test_metadata_server_restart(self):
+        :avocado: tags=all,full_regression
+        :avocado: tags=nvme,large,hw
+        :avocado: tags=metadata
+        :avocado: tags=metadata_no_leader_change
         """
-        self.pool.pool.connect(2)
-        init_container = NO_OF_MAX_CONTAINER
-        additional_container = 1000000
-        der_no_space = "RC: -1007"
+        for k in range(10):
+            self.log.info("Container create/delete cycle %d / 9", k)
+            pool_leader = self.get_pool_leader()
+            self.log.info("Current leader: %s", pool_leader)
 
-        container_array = []
-        self.log.info("(1)Start Creating %d containers..", init_container)
-        for cont in range(init_container):
-            container = DaosContainer(self.context)
-            container.create(self.pool.pool.handle)
-            container_array.append(container)
+            container_array, _ = self.create_containers_until_error()
+
+            if pool_leader != self.get_pool_leader():
+                self.fail("Leader unexpectedly changed after creating conts.")
+
+            self.delete_containers_in_array(container_array)
+
+            if pool_leader != self.get_pool_leader():
+                self.fail("Leader unexpectedly changed after deleting conts.")
+        self.log.info("Test passed, no error found within create/delete cycle.")
+
+    def test_metadata_pool_replica_leadership_change_no_space(self):
+        """
+        Test description:
+            Verify a pool leadership changes correctly when metadata is
+            filled up.
+
+        Test steps:
+            Create a pool with storage replica(s) (using --nranks).
+            Query pool and save rank leader.
+            Fill metadata until DER_NOSPACE.
+            Change leadership (at the moment it should be done by stopping
+            the server that has the leader rank, for easy use, use 1 rank per
+            server).
+            Wait for rebuild to start -> complete.
+            Verify the pool is still serviceable:
+                Query pool
+                Attempt to add containers until DER_NOSPACE
+                Delete containers
+                Add containers again
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=small,hw
+        :avocado: tags=metadata
+        :avocado: tags=metadata_leader_change_no_space
+        """
+        dmg = self.get_dmg_command()
+        pool_leader = self.get_pool_leader()
+        self.log.info("Current leader: %s", pool_leader)
 
-        self.log.info("(1.1)%d Container Created.", init_container)
-        self.log.info("(1.2)Create additional 100 Containers, expect Fail.")
+        container_array, container_length = self.create_containers_until_error()
 
-        #Create additional containers to check for DaosApiError -1007
-        for cont in range(additional_container):
+        dmg.system_stop(force=True, ranks=pool_leader)
+
+        # Wait for pool rebuild
+        if self.get_pool_rebuild_status == "idle":
```

Comment:
(pylint-comparison-with-callable) Comparing against a callable, did you omit the parenthesis?